File: core/frontend/static/lib/utils/strip-outer-markdown-fence.js
/**
 * Если вся строка — один ограждающий GFM-блок (язык markdown/md или без метки),
 * возвращает содержимое внутри. Иначе — исходную строку без изменений.
 */
export function stripOuterMarkdownCodeFence(raw) {
    if (typeof raw !== 'string') {
        return '';
    }
    const trimmed = raw.trim();
    if (trimmed.length === 0) {
        return '';
    }
    const openMatch = trimmed.match(/^```(?:markdown|md)?\s*\r?\n/i);
    if (!openMatch) {
        return trimmed;
    }
    const afterOpen = trimmed.slice(openMatch[0].length);
    const closeIdx = afterOpen.lastIndexOf('```');
    if (closeIdx < 0) {
        return trimmed;
    }
    const inner = afterOpen.slice(0, closeIdx).trim();
    return inner.length === 0 ? trimmed : inner;
}

File: core/text_transforms/strip_outer_markdown_fence.py
"""Снятие одиночной внешней GFM-ограды ```markdown/``` с текста (ответы LLM)."""

from __future__ import annotations

import re


def strip_outer_markdown_code_fence(raw: str) -> str:
    trimmed = raw.strip()
    if not trimmed:
        return ""
    open_m = re.match(r"^```(?:markdown|md)?\s*\r?\n", trimmed, re.IGNORECASE)
    if not open_m:
        return trimmed
    after_open = trimmed[open_m.end() :]
    close_idx = after_open.rfind("```")
    if close_idx < 0:
        return trimmed
    inner = after_open[:close_idx].strip()
    return inner if inner else trimmed
